src/lib/y2storage/devices_lists/base.rb

```diff
@@ -76,10 +76,17 @@ def dup
         self.class.new(devicegraph, @list.dup)
       end
 
+      def +(other)
+        if self.class != other.class
```

Comment:
Intentional. We always want our lists to contain elements of a given Storage class. Classes coming from Storage can't deal with duck typing, so we don't want to mix them.